tabular/src/autogluon/tabular/models/tabpfn/tabpfn_model.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import pandas as pd
+
+from autogluon.features.generators import LabelEncoderFeatureGenerator
+from autogluon.core.constants import BINARY, MULTICLASS
+from autogluon.core.models import AbstractModel
+from autogluon.core.utils import generate_train_test_split
+
+
+# TODO: Implement max_sets for bagging as a model config option, consider `max_sets=1` for TabPFN by default,
+#  to avoid unreasonable inference times.
+class TabPFNModel(AbstractModel):
+    """
+    AutoGluon model wrapper to the TabPFN model: https://github.com/automl/TabPFN
+
+    Paper: "TabPFN: A Transformer That Solves Small Tabular Classification Problems in a Second"
+    Authors: Noah Hollmann, Samuel Müller, Katharina Eggensperger, and Frank Hutter
+
+    TabPFN is a viable model option when inference speed is not a concern,
+    and the number of rows of training data is less than 10,000.
+
+    Additionally, TabPFN is only available for classification tasks with up to 10 classes and 100 features.
+
+    To use this model, `tabpfn` must be installed.
+    To install TabPFN, you can run `pip install autogluon.tabular[tabpfn]` or `pip install tabpfn`.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    def _fit(self, X: pd.DataFrame, y: pd.Series, **kwargs):
+        from tabpfn import TabPFNClassifier
+        ag_params = self._get_ag_params()
+        sample_rows = ag_params.get('sample_rows')
+        max_features = ag_params.get('max_features')
+        max_classes = ag_params.get('max_classes')
+        if max_classes is not None and self.num_classes > max_classes:
+            # TODO: Move to earlier stage when problem_type is checked
+            raise AssertionError(f'Max allowed classes for the model is {max_classes}, '
+                                 f'but found {self.num_classes} classes.')
+
+        # TODO: Make sample_rows generic
+        if sample_rows is not None and len(X) > sample_rows:
+            X, y = self._subsample_train(X=X, y=y, num_rows=sample_rows)
```

Comment:
We have a bit of data on this. The best strategy is to something like have multiple trees with a TabPFN in each leaf, but this is not too much worse in our experience.

Reply:
That makes sense, I had a similar assumption as to what would probably work best.

For now, I've kept as is since doing the `TabPFN Forest` approach is very similar to using bagging in AutoGluon on `TabPFN`. I'd expect `TabPFN_Forest` to be slightly better, but will keep things simple for now. If you do have data that shows a compelling performance improvement I'd definitely take another look.

If `TabPFN` can do this logic internally, that could be a very nice addition. For example, when `N_ensemble_configurations=10`, each TabPFN in N could use a different sample from the train data input, which should maximize diversity.